packages/flutter/test/cupertino/nav_bar_transition_test.dart

```diff
@@ -199,29 +201,80 @@ void main() {
         tester.renderObject(flying(tester, find.text('Page 1')).last);
     expect(topBackLabel.text.style.color, const Color(0xFF00070F));
     expect(topBackLabel.text.style.fontWeight, FontWeight.w600);
-    expect(topBackLabel.text.style.fontFamily, '.SF UI Text');
-    expect(topBackLabel.text.style.letterSpacing, -0.08952957153320312);
+    expect(topBackLabel.text.style.fontFamily, '.SF Pro Text');
+    expect(topBackLabel.text.style.letterSpacing, -0.41);
 
     checkOpacity(tester, flying(tester, find.text('Page 1')).last, 0.0);
 
     // Move animation further a bit.
     await tester.pump(const Duration(milliseconds: 200));
     expect(bottomMiddle.text.style.color, const Color(0xFF0073F0));
     expect(bottomMiddle.text.style.fontWeight, FontWeight.w400);
-    expect(bottomMiddle.text.style.fontFamily, '.SF UI Text');
-    expect(bottomMiddle.text.style.letterSpacing, -0.231169798374176);
+    expect(bottomMiddle.text.style.fontFamily, '.SF Pro Text');
+    expect(bottomMiddle.text.style.letterSpacing, -0.41);
 
     checkOpacity(tester, flying(tester, find.text('Page 1')).first, 0.0);
 
     expect(topBackLabel.text.style.color, const Color(0xFF0073F0));
     expect(topBackLabel.text.style.fontWeight, FontWeight.w400);
-    expect(topBackLabel.text.style.fontFamily, '.SF UI Text');
-    expect(topBackLabel.text.style.letterSpacing, -0.231169798374176);
+    expect(topBackLabel.text.style.fontFamily, '.SF Pro Text');
+    expect(topBackLabel.text.style.letterSpacing, -0.41);
 
     checkOpacity(
         tester, flying(tester, find.text('Page 1')).last, 0.8733493089675903);
   });
 
+  testWidgets('Font transitions respect themes',
+      (WidgetTester tester) async {
+      await startTransitionBetween(
+        tester,
+        fromTitle: 'Page 1',
+        theme: const CupertinoThemeData(brightness: Brightness.dark),
+      );
+
+      // Be mid-transition.
+      await tester.pump(const Duration(milliseconds: 50));
+
+      // The transition's stack is ordered. The bottom middle is inserted first.
+      final RenderParagraph bottomMiddle =
+      tester.renderObject(flying(tester, find.text('Page 1')).first);
+      expect(bottomMiddle.text.style.color, const Color(0xFFFFF8EF));
+      expect(bottomMiddle.text.style.fontWeight, FontWeight.w600);
+      expect(bottomMiddle.text.style.fontFamily, '.SF Pro Text');
+      expect(bottomMiddle.text.style.letterSpacing, -0.41);
+
+      checkOpacity(
+        tester, flying(tester, find.text('Page 1')).first, 0.8609542846679688);
```

Comment:
This doesn't seem likely to be a stable value to test. Maybe use [closeTo](
https://docs.flutter.io/flutter/package-matcher_matcher/closeTo.html) here or incorporate closeTo in `checkOpacity`.

Reply:
I thought it might be flaky too (there's a couple of other tests that checks doubles in this file) but it seems to have never failed on any platform. 

What are the source of the flakiness for something like this typically?

Reply:
Differences in floating point implementations seem like one source of hard to understand failures.

Reply:
Added moreOrLessEquals in checkOpacity